backend/corpora/common/entities/dataset_asset.py

```diff
@@ -0,0 +1,50 @@
+import logging
+import typing
+
+import boto3
+from botocore.exceptions import ClientError
+
+from .entity import Entity
+from ..corpora_orm import DbDatasetArtifact
+
+logger = logging.getLogger(__name__)
+
+
+class DatasetAsset(Entity):
+    table = DbDatasetArtifact
+    s3 = boto3.client("s3")
+
+    def __init__(self, db_object: DbDatasetArtifact):
+        super().__init__(db_object)
+
+        self.bucket_name, self.file_prefix = self.s3_uri[5:].split("/", 1)
```

Comment:
Also this it a bit of a nit, but this talks about `file_prefix` a lot, but I think it's really a `key name` or `object key`. The distinction is kind of important because we have things like `s3://corpora/project_id/dataset_id/dataset.loom` which will work with these methods because it is a key. But we also have `s3://corpora/project_id/dataset_id/dataset.cxg/` which will not work because it is just a prefix.

Reply:
I like it, I'll make the change.

Reply:
> You could maybe use urlparse so you can assert that the scheme is s3

Shouldn't this be caught during ingestion? If not what kind of error would we return, server error?